src/rpi.py

```diff
@@ -0,0 +1,132 @@
+import binascii
+import logging
+import subprocess
+import os
+import shutil
+import tempfile
+import hashlib
+from datetime import datetime
+
+from utils import RPI_DIR
+
+LOG = logging.getLogger("rpi")
+
+
+def keys():
+    dirs = os.listdir(RPI_DIR)
+    response = {
+        "keys": [dirs],
+    }
+    return response
+
+
+def new(body, user):
+    tmpdir = tempfile.mkdtemp(prefix="rpi-")
+    cmd = [ "openssl", "genpkey", "-algorithm", "RSA", "-out",
+           tmpdir + os.sep + "private.pem", "-pkeyopt",
+           "rsa_keygen_bits:2048" ]
+    cmd_result = subprocess.run(cmd, stdout=subprocess.DEVNULL,
+                                stderr=subprocess.DEVNULL, check = False)
+    if cmd_result.returncode != 0:
+        raise RuntimeError("Failed to generate RPI key pair")
+
+    cmd = [ "openssl", "rsa", "-pubout", "-in", tmpdir + os.sep + "private.pem", "-out",
```

Comment:
Nitpick: `os.path.join(tmpdir, "private.pem")` would be preferred to `tmpdir + os.sep + "private.pem"` in python, throughout the file.